great_expectations/datasource/fluent/interfaces.py

```diff
@@ -1085,7 +1085,9 @@ def _validator(self) -> V1Validator:
                 "We can't validate batches that are attached to datasources without a data context"
             )
         batch_definition = self.data_asset.add_batch_definition(
```

Comment:
I'm stressed that we are adding a new batch_definition here. I want to put up a rival PR to see if we can solve the bug and also avoid this side-effect.